src/panels/config/automation/trigger/types/ha-automation-trigger-calendar.ts

```diff
@@ -0,0 +1,73 @@
+import { html, LitElement } from "lit";
+import { customElement, property } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import type { CalendarTrigger } from "../../../../../data/automation";
+import type { HomeAssistant } from "../../../../../types";
+import type { TriggerElement } from "../ha-automation-trigger-row";
+import type { HaFormSchema } from "../../../../../components/ha-form/types";
+import type { LocalizeFunc } from "../../../../../common/translations/localize";
+
+@customElement("ha-automation-trigger-calendar")
+export class HaCalendarTrigger extends LitElement implements TriggerElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public trigger!: CalendarTrigger;
+
+  private _schema = memoizeOne((localize: LocalizeFunc) => [
+    {
+      name: "entity_id",
+      required: true,
+      selector: { entity: { domain: "calendar" } },
+    },
+    {
+      name: "event",
+      type: "select",
+      required: true,
+      options: [
+        [
+          "start",
+          localize(
+            "ui.panel.config.automation.editor.triggers.type.calendar.start"
+          ),
+        ],
+      ],
+    },
```

Comment:
Is the goal to have more than start here?

Reply:
Yes, i have an "end" PR as a followup. I maybe should have just added it to the original core PR, but was trying to keep the first PR simple.

I will also add an "offset" too after that.

Reply:
Sounds good 👍  I converted to draft so when you are ready just set it to ready for review and we'll take a look

Reply:
We can merge the frontend PR with just Start for now